nixos/modules/system/activation/switch-to-configuration.pl

```diff
@@ -20,6 +20,10 @@
 
 my $action = shift @ARGV;
 
+# Prepare the environment
+foreach (keys %ENV) {
+    delete $ENV{$_};
+}
```

Comment:
```suggestion
undef %ENV;
```

Reply:
That won't work because we both forgot about the `NIXOS_NO_SYNC` variable which needs to be retained